### PR TITLE
BP-967 Create support for Reward point functionality

### PR DESCRIPTION
**Pull request by saranchuk-hys**

### CODE REVIEW

File: Model/Method/AbstractMethod.php [hunk 1]
```diff
@@ -2430,5 +2432,38 @@ public function getShippingAddress($payment, &$isAbsent = false)
...
+            }
+        }
+
+        foreach ($requestData2 as $key=>$item){
+            $itemsTotalAmount += $item['Quantity'] * $item[$this->getPriceFieldName()];
+            $count++; 
+        }
+
+        //Add diff line
+        if(abs($order->getBaseGrandTotal() - $itemsTotalAmount) > 0.01){
```
Comment (vladislav-padalka-hys) [inline Model/Method/AbstractMethod.php:2459]: what about usage of areEqualAmounts from helper ?

File: Model/Method/AbstractMethod.php [hunk 2]
```diff
@@ -2430,5 +2432,38 @@ public function getShippingAddress($payment, &$isAbsent = false)
...
+                if($item['Name'] == 'Quantity'){
+                    $requestData2[$item['GroupID']]['Quantity'] = $item['_'];
+                }
+                if($item['Name'] == $this->getPriceFieldName()){
+                    $requestData2[$item['GroupID']][$this->getPriceFieldName()] = $item['_'];
+                }
+            }
+        }
+
+        foreach ($requestData2 as $key=>$item){
```
Comment (vladislav-padalka-hys) [inline Model/Method/AbstractMethod.php:2453]: could you please adjust spaces indentation in this new function

**Review by vladislav-padalka-hys — CHANGES_REQUESTED** [review]

File: Model/Method/AbstractMethod.php (hunk 1 above)
Comment (saranchuk-hys) [reply Model/Method/AbstractMethod.php]: sure, fixed, thank you

File: Model/Method/AbstractMethod.php (hunk 2 above)
Comment (saranchuk-hys) [reply Model/Method/AbstractMethod.php]: sure, fixed, thank you

**Review by vladislav-padalka-hys — APPROVED** [review]

**Pull request merged**